src/components/text-area/text-area.tsx

```diff
@@ -0,0 +1,476 @@
+  }
+
+  disconnectedCallback(): void {
+    disconnectLabel(this);
+    disconnectForm(this);
+    disconnectLocalized(this);
+    disconnectMessages(this);
+    this.resizeObserver?.disconnect();
+  }
+
+  render(): VNode {
+    const hasFooter = this.startSlotHasElements || this.endSlotHasElements || !!this.maxLength;
+    return (
+      <Host>
+        <textarea
+          aria-invalid={toAriaBoolean(this.value?.length > this.maxLength)}
+          aria-label={getLabelText(this)}
+          autofocus={this.autofocus}
+          class={{
+            [CSS.readOnly]: this.readOnly,
+            [CSS.textAreaInvalid]: this.value?.length > this.maxLength,
+            [CSS.footerSlotted]: this.endSlotHasElements && this.startSlotHasElements,
+            [CSS.blockSizeFull]: !hasFooter,
+            [CSS.borderColor]: !hasFooter
+          }}
+          cols={this.columns}
+          disabled={this.disabled}
+          name={this.name}
+          onChange={this.handleChange}
+          onInput={this.handleInput}
+          placeholder={this.placeholder}
+          readonly={this.readOnly}
+          required={this.required}
+          rows={this.rows}
+          value={this.value}
+          wrap={this.wrap}
+          // eslint-disable-next-line react/jsx-sort-props
+          ref={this.setTextAreaEl}
+        />
+        <span class={{ [CSS.content]: true }}>
+          <slot onSlotchange={this.contentSlotChangeHandler} />
+        </span>
+        <footer
+          class={{
+            [CSS.footer]: true,
+            [CSS.readOnly]: this.readOnly,
+            [CSS.hide]: !hasFooter
+          }}
+          ref={(el) => (this.footerEl = el as HTMLElement)}
+        >
+          <div
+            class={{
+              [CSS.container]: true,
+              [CSS.footerEndSlotOnly]: !this.startSlotHasElements && this.endSlotHasElements
+            }}
+          >
+            <slot
+              name={SLOTS.footerStart}
+              onSlotchange={(event) =>
+                (this.startSlotHasElements = slotChangeHasAssignedElement(event))
+              }
+            />
+            <slot
+              name={SLOTS.footerEnd}
+              onSlotchange={(event) =>
+                (this.endSlotHasElements = slotChangeHasAssignedElement(event))
+              }
+            />
+          </div>
+          {this.renderCharacterLimit()}
+        </footer>
+        <HiddenFormInputSlot component={this} />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.textAreaEl.focus();
+  }
+
+  /** Selects the text of the component's `value`. */
+  @Method()
+  async selectText(): Promise<void> {
+    this.textAreaEl.select();
```

Comment:
Should `selectText`  await `componentLoaded(this)` like `setFocus` does?

Reply:
i think it should.